### PR TITLE
Add arm support to docker images

### PR DESCRIPTION
**Pull request by bwoebi**
### Description

To be built with `docker buildx bake`.

This pull request adds arm support to our docker images. In particular it:
- Sets platform configurations in our docker-compose files
- Uses cross-compilation for our buster images (which, on average get rebuild more often, also for quicker testing of new things). Other docker images use qemu, which is slower, but still reasonable. (Collapsing the buster PHP builds into a couple files to not have too much copy paste per version to maintain.)
- Streamlines the randomized tests images to depend on our centos php images, as to avoid setting up PHP compilation twice (hence the changes to centos configure, so that it's also usable in randomized tests).

This is the foundational work, creating docker images for arm, to be able to actually run them in CI. This will be a separate pull request.

### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ x-aliases:
         - agent
         - ddagent_integration
         - request-replayer
-        - elasticsearch2_integration
+        #- elasticsearch2_integration
```
Comment (bwoebi) [inline docker-compose.yml:20]: Not sure about this, but there's no elasticsearch2 container for arm... Leaving this here will just completely block the containers from running on arm, so I think we can leave them commented out?

File: dockerfiles/ci/buster/0001-Better-support-for-cross-compilation.patch [hunk 2]
```diff
@@ -0,0 +1,288 @@
+From f1ad9199eff32dd805bb5b155002bdb6ca69bf01 Mon Sep 17 00:00:00 2001
```
Comment (krakjoe) [inline dockerfiles/ci/buster/0001-Better-support-for-cross-compilation.patch:1]: Just out of interest, and so there's a record somewhere ...

Is this included in any version upstream ? If not, why not ?

**Review by krakjoe — APPROVED** [review]
This all looks pretty sane to me, not able to test it directly as not blessed with arm yet :(

File: docker-compose.yml (hunk 1 above)
Comment (krakjoe) [reply docker-compose.yml:20]: For now, this seems okay.

File: dockerfiles/ci/buster/0001-Better-support-for-cross-compilation.patch (hunk 2 above)
Comment (bwoebi) [reply dockerfiles/ci/buster/0001-Better-support-for-cross-compilation.patch:1]: This patch has been applied to 8.1, but not to 8.0 on upstream. We only need it for cross-compiling 8.0. So maybe something which we can backport directly to 8.0...

**Pull request merged**